core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -10791,7 +10791,7 @@ public User getUserWithID(String userID, String[] requestedClaims, String profil
                     realmConfig.getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME));
             String errorCode = ErrorMessages.ERROR_CODE_NON_EXISTING_USER.getCode();
             handleGetUserFailureWithID(errorCode, errorMessage, userID, requestedClaims, profileName);
-            throw new UserStoreException(errorCode + " - " + errorMessage);
+            throw new UserStoreException(errorCode + " - " + errorMessage,errorCode);
```

Comment:
formatting issue

Reply:
Do we need to have errorCode in the message as well?

Reply:
@IsuraD is there any possibility that some places might have check the string message for the error code?

Reply:
There are a lot of places other than this in AbstractUserStoreManager class which throws the UserStoreException in the same manner.  We need to fix all the places. In that case, it is safe to keep the errorCode in the message.

Reply:
[UniqueIDToUser](https://github.com/wso2/identity-api-user/blob/v1.0.64/components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/function/UniqueIdToUser.java#L79) need to throw APIError with status Not_Found. In this case we need extract error code from the message.